### DISCUSSION TITLE
Installing later versions of node for the data-visuals kit

### DISCUSSION THREAD

**Discussion by CarlaAstudillo** [texastribune/data-visuals-guides#27] [General]
In the instructions, we r[ecommend installing](https://github.com/texastribune/data-visuals-guides/blob/3fc258bca8fcd5226075b78dce22db21d8b849de/computer-setup.md?plain=1#L181) very old LTS versions node that I personally don't use anymore:

- lts/carbon (v8.17.0)
- lts/dubnium (v10.24.1)
- lts/erbium (v12.22.12)

I currently use v18.9.0 as my default node to use the data-visuals-kit so I wonder if we should bump up the versions of LTS we put in the instructions to use.

- lts/gallium (v16.20.2)
- lts/hydrogen (v18.20.5)

We can also maybe make it optional for new people to `nvm install` older versions of node unless they want to open up older projects (we can give them a list of past node versions that had been default that they can install)

In addition, when using `npx` to make new graphics in the kit, it keeps telling me to update my npm (and therefore my node). 
<img width="467" alt="Screenshot 2024-12-17 at 4 14 41 PM" src="https://github.com/user-attachments/assets/dd1ad906-6914-4f56-abad-a8298c3cc3bb" />

It says it's a major update so I'm a little nervous that using later versions might break the kit? I can test it out but might look into updating the [data-visuals kit](https://github.com/texastribune/data-visuals-create) anyway to make sure it's working with newer versions of node. 

The latest LTS node version that you can download through nvm is `lts/jod -> v22.12.0`.

**Comment by csessig86** [comment]
What if instead of installing those old node versions, we have them install the latest LTS [release](https://stackoverflow.com/a/65532065):
`nvm install --lts`
`nvm use --lts`

I tested this and didn't run into issues working on a graphic created using the kit. Looping in @Yuriko-Schumacher. What do you all think? @CarlaAstudillo maybe we don't need to do anything to get the kit working with newer versions of node? Unless you had specific ideas in mind.

**Reply by CarlaAstudillo** [reply]
I actually used the latest node version on the House speaker vote graphic and I didn't run into any issues either, so we might actually be good?

And I'm all for just installing the latest versions but we should still keep a reminder to download older versions if you're trying to open up projects that are years old.

**Reply by csessig86** [reply]
Yeah definitely, that's why I think the instructions to install `nvm` is still worth keeping in there.

**Reply by Yuriko-Schumacher** [reply]
I updated to node v23 and found no issues by creating a project with the kit. So I think for now we can default it to install the latest version and make sure to use older versions for older projects. Current kit technically supports any node versions 17 or later. So for newer projects, node just can't be v16 or older.

I will find time to update kit soon and try addressing some other warnings and Puppeteer issues!
